packages/x-data-grid-premium/src/components/promptControl/GridToolbarPromptControl.tsx

```diff
@@ -19,12 +19,12 @@ import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
 import { PromptResponse } from '../../hooks/features/promptControl/types';
 import { RecordButton, BrowserSpeechRecognition } from './RecordButton';
 
-type OwnerState = DataGridPremiumProcessedProps;
+type OwnerState = Pick<DataGridPremiumProcessedProps, 'classes'> & { recording: boolean };
 
 const supportsSpeechRecognition = !!BrowserSpeechRecognition;
 
-const useUtilityClasses = (ownerState: OwnerState, recording: boolean) => {
-  const { classes } = ownerState;
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes, recording } = ownerState;
```

Comment:
For context, I wrote it like this because `ownerState` in theory should be the owner component props (the DataGrid being the owner component this slot belongs to), and also if we can avoid creating more objects that's always a good thing perf-wise (though it doesn't matter in practice because this component is only rendered once). If this was a cell or something rendered a lot, I would argue to revert this change, but it doesn't matter enough here.